src/LibraryManager.Vsix/Shared/LibraryCommandService.cs

```diff
@@ -0,0 +1,352 @@
+        {
+            CancelOperation();
+        }
+
+        private void OnBeforeCloseSolution(object sender, ParamEventArgs e)
+        {
+            CancelOperation();
+        }
+
+        public bool IsOperationInProgress
+        {
+            get
+            {
+                lock (_lockObject)
+                {
+                    return _currentOperationTask != null && !_currentOperationTask.IsCompleted;
+                }
+            }
+        }
+
+        public async Task RestoreAsync(string configFilePath, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            Dictionary<string, Manifest> manifests = await GetManifestFromConfigAsync(new[] { configFilePath }, cancellationToken);
+            await RestoreAsync(manifests, cancellationToken);
+        }
+
+        public async Task RestoreAsync(IEnumerable<string> configFilePaths, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            Dictionary<string, Manifest> manifests = await GetManifestFromConfigAsync(configFilePaths, cancellationToken);
+            await RestoreAsync(manifests, cancellationToken);
+        }
+
+        public async Task RestoreAsync(string configFilePath, Manifest manifest, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            await RestoreAsync(new Dictionary<string, Manifest>() { [configFilePath] = manifest }, cancellationToken);
+        }
+
+        public async Task UninstallAsync(string configFilePath, string libraryId, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            string taskTitle = GetTaskTitle(OperationType.Uninstall, libraryId);
+            string errorMessage = string.Format(LibraryManager.Resources.Text.Uninstall_LibraryFailed, libraryId);
+
+            await RunTaskAsync((internalToken) => UninstallLibraryAsync(configFilePath, libraryId, internalToken), taskTitle, errorMessage);
+        }
+
+        public async Task CleanAsync(ProjectItem configProjectItem, CancellationToken cancellationToken = default(CancellationToken))
+        {
+
+            string taskTitle = GetTaskTitle(OperationType.Clean, string.Empty);
+
+            await RunTaskAsync((internalToken) => CleanLibrariesAsync(configProjectItem, internalToken), taskTitle, LibraryManager.Resources.Text.Clean_OperationFailed);
+        }
+
+        private async Task RestoreAsync(Dictionary<string, Manifest> manifests, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            string taskTitle = GetTaskTitle(OperationType.Restore, string.Empty);
+            string errorMessage = LibraryManager.Resources.Text.Restore_OperationFailed;
+
+            await RunTaskAsync((internalToken) => RestoreInternalAsync(manifests, internalToken), taskTitle, errorMessage);
+        }
+
+        private async Task RunTaskAsync(Func<CancellationToken, Task> toRun, string taskTitle, string errorMessage)
```

Comment:
Could probably come up with a better name than toRun for this. getTask or something similar

Reply:
Fixed